### PR TITLE
Add calico v3.22.1 package

### LINKED ISSUE #3743: Add calico 3.22.1 to addons
## Feature Request
[Calico 3.22.1](https://projectcalico.docs.tigera.io/archive/v3.22/about/about-calico) is released. Add calico 3.22.1 package.

### PR DESCRIPTION
**Pull request by 12345lcr**
## What this PR does / why we need it
* Added calico v3.22.1 package
* Added tests for calico v3.22.1 package
* Removed calico-ce package which is also v3.22.1

## Which issue(s) this PR fixes
Fixes: #3743

## Describe testing done for PR
Ran make test for Calico 3.22.1 package.

### CODE REVIEW

**Comment by 12345lcr** [comment]
Hi @joshrosso @stmcginnis , for pr https://github.com/vmware-tanzu/community-edition/pull/3550, I removed this calico-ce package and integrate v3.22.1 back to calico packages, please take a look. If there is any background that we need to keep 2 kinds of calico packages please let me know, thanks!

And perhaps since the v3.22.1 package bundle is already pushed by calico-ce package, when I push the package to the registry, the error below shows. So I am temporarily using a personal registry. Not sure how to handle this, cc @seemiller .

```
Error: Writing 'projects.registry.vmware.com/tce/calico:3.22.1': Writing image: Retried 5 times: PUT https://projects.registry.vmware.com/v2/tce/calico/manifests/3.22.1: PRECONDITION: Failed to process request due to 'calico:3.22.1' configured as immutable.
```

**Comment by 12345lcr** [comment]
The reason that this pr involves harbor package owners is because I removed one blank line in harbor package tests, which was blocking pr lint. No other modifications made.

File: addons/packages/calico/3.22.1/bundle/config/overlay/calico_overlay.yaml [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    kapp.k14s.io/update-strategy: fallback-on-replace
+
+#@overlay/match by=overlay.subset({"kind":"DaemonSet", "metadata": {"name": "calico-node"}})
+---
+kind: DaemonSet
+metadata:
+  #@overlay/match missing_ok=True
+  annotations:
+    kapp.k14s.io/disable-default-label-scoping-rules: ""
+    kapp.k14s.io/update-strategy: fallback-on-replace
```
Comment (12345lcr) [inline addons/packages/calico/3.22.1/bundle/config/overlay/calico_overlay.yaml:26]: Hi @vijaykatam , in this v3.22.1, I followed your suggestion to change the update strategy to `fallback-on-replace` for both calico daemonset and calico deployment, feel free to take a look and leave any comments, thanks.

File: addons/packages/calico/3.22.1/package.yaml [hunk 2]
```diff
@@ -51,7 +46,7 @@ spec:
       syncPeriod: 5m
       fetch:
       - imgpkgBundle:
-          image: projects.registry.vmware.com/tce/calico@sha256:31c240996c50b1fd06339987465a95b01a982050fff1ecd0262e8ae1e50206e4
+          image: index.docker.io/licrui/tce-calico@sha256:a4ccb6a2a6871f2b1d24a04416f0e99d6bb1f295c6c0eeee0546992470b57840
```
Comment (seemiller) [inline addons/packages/calico/3.22.1/package.yaml:49]: Can you push this to `projects.registry.vmware.com/tce`?

**Review by seemiller — CHANGES_REQUESTED** [review]
Anytime a file changes in the `bundle/config` directory a new bundle should be pushed to our Harbor project, `projects.registry.vmware.com/tce`. Could you you please do this for the 3.11.3, 3.19.1 and 3.22.1 versions and update the respective package.yaml files with the update SHA/digest.

File: addons/packages/calico/3.22.1/bundle/config/overlay/calico_overlay.yaml (hunk 1 above)
Comment (seemiller) [reply addons/packages/calico/3.22.1/bundle/config/overlay/calico_overlay.yaml:26]: @12345lcr Sorry, just re-read your comment above. You can use the `push-package` make task in the TCE repo, just use a tag like `3.22.1-2`. We've enabled immutable tags, so we can no longer re-use tags.

File: addons/packages/calico/3.22.1/package.yaml (hunk 2 above)
Comment (seemiller) [reply addons/packages/calico/3.22.1/package.yaml:49]: @12345lcr Sorry, just re-read your comment above. You can use the `push-package` make task in the TCE repo, just use a tag like `3.22.1-2`. We've enabled immutable tags, so we can no longer re-use tags.

```
make push-package PACKAGE=calico VERSION=3.22.1 TAG=3.22.1-2
```

**Comment by 12345lcr** [comment]
Hi @seemiller , I removed the schema part and pushed to TAG=3.22.1-3. Because this schema implementation seems too strict and have backward compatibility issue. I am about to drive a discussion in tce and tanzu-framework, and once the decision is made to add schemas, I will create another issue to track the schema implementation for calico 3.22.1. 

For now the pr to bump calico 3.22.1 is ready for reviewing, feel free to take a look.

**Review by wjun — APPROVED** [review]
LGTM

**Review by seemiller — APPROVED** [review]

**Pull request merged**